### PR TITLE
Virtualization Detail Tab Page

### PR DESCRIPTION
**Pull request by tejones**
Implements the detail page for a virtualization. Each tab on the page will have it's own page. The default tab/page is Views. The other tabs have stubbed out pages with this commit and will need to be implemented.

TODOs:
- Select the View tab on load
- Fix tab links to set the virtualizationId when tabs are selected
- Add state value to HeaderView
- Create Storybook for ViewPage

### CODE REVIEW

**Comment by riccardo-forina** [comment]
PR Storybook available [here](https://circleci.com/api/v1.1/project/github/syndesisio/syndesis-react-poc/391/artifacts/0/home/circleci/project/doc/index.html)

**Comment by riccardo-forina** [comment]
PR Storybook available [here](https://circleci.com/api/v1.1/project/github/syndesisio/syndesis-react-poc/393/artifacts/0/home/circleci/project/doc/index.html)

**Comment by riccardo-forina** [comment]
PR Storybook available [here](https://circleci.com/api/v1.1/project/github/syndesisio/syndesis-react-poc/395/artifacts/0/home/circleci/project/doc/index.html)

**Comment by tejones** [comment]
> I think it looks good, but would like @riccardo-forina to take a look...
> On the refactoring, I'd prefer to stay with a more shallow heirarchy (eg Virtualizations/Views/ViewList.tsx instead of Virtualizations/Detail/Views/ViewList.tsx) - Unless there is a strong reason to leave in the 'Detail' folder.

I am good with that hierarchy @mdrillin. I'll update the PR.

**Comment by tejones** [comment]
> > I think it looks good, but would like @riccardo-forina to take a look...
> > On the refactoring, I'd prefer to stay with a more shallow heirarchy (eg Virtualizations/Views/ViewList.tsx instead of Virtualizations/Detail/Views/ViewList.tsx) - Unless there is a strong reason to leave in the 'Detail' folder.
> 
> I am good with that hierarchy @mdrillin. I'll update the PR.

PR updated to remove details and move Views to be a child of virtualizations.

**Comment by riccardo-forina** [comment]
PR Storybook available [here](https://circleci.com/api/v1.1/project/github/syndesisio/syndesis-react-poc/399/artifacts/0/home/circleci/project/doc/index.html)

File: packages/api/src/WithViews.tsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+/**
+ * A component that fetches the virtualization views with the specified identifier.
+ * @see [virtualizationId]{@link IWithViewProps#virtualizationId}
+ */
+export class WithViews extends React.Component<IWithViewProps> {
+  public render() {
+    return (
+      <DVFetch<RestViewDefinition[]>
+        url={`workspace/vdbs/${
+          this.props.virtualizationId
```
Comment (riccardo-forina) [inline packages/api/src/WithViews.tsx]: `virtualizationId` could be undefined accordingly to the interface, which would generate an url like `workspace/vdbs/undefinedvdb/Models/views/Views`. Is this correct?

File: packages/api/src/WithVirtualization.tsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+ * A component that fetches the virtualization with the specified identifier.
+ * @see [virtualizationId]{@link IWithVirtualizationProps#virtualizationId}
+ */
+export class WithVirtualization extends React.Component<
+  IWithVirtualizationProps
+> {
+  public render() {
+    return (
+      <DVFetch<RestDataService>
+        url={`workspace/dataservices/${this.props.virtualizationId}`}
```
Comment (riccardo-forina) [inline packages/api/src/WithVirtualization.tsx]: Same comment as before here, `virtualizationId` could be undefined. Is this right?

File: packages/api/src/WithViews.tsx (hunk 1 above)
Comment (tejones) [reply packages/api/src/WithViews.tsx]: Good catch @riccardo-forina. That should not be optional.

File: packages/api/src/WithVirtualization.tsx (hunk 2 above)
Comment (tejones) [reply packages/api/src/WithVirtualization.tsx]: That should not be optional here either. I will amend. Thanks.

**Comment by riccardo-forina** [comment]
PR Storybook available [here](https://circleci.com/api/v1.1/project/github/syndesisio/syndesis-react-poc/401/artifacts/0/home/circleci/project/doc/index.html)

**Pull request merged**